docs/docs/concepts/breakpoints.md

```diff
@@ -0,0 +1,281 @@
+# Breakpoints
+
+Breakpoints pause graph execution at specific points, enabling [**human-in-the-loop**](./human_in_the_loop.md) workflows and debugging. Breakpoints are powered by  LangGraph's [**persistence layer**](./persistence.md), which saves the state after each graph step. 
+
+## Requirements
+
+To use breakpoints, you will need to:
+
+1. [**Specify a checkpointer**](persistence.md#checkpoints) to save the graph state after each step.
+2. [**Set breakpoints**](#setting-breakpoints) to specify where execution should pause.
+3. Run the graph with a [**thread ID**](./persistence.md#threads) to pause execution at the breakpoint.
+4. [**Resume execution**](#resuming) from the paused state.
+
+## Setting breakpoints
+
+There are two places where you can set breakpoints:
+
+1. **Inside** a node using the [`interrupt` function](#the-interrupt-function) (or the older [`NodeInterrupt` exception](#nodeinterrupt-exception)).
```

Comment:
should we still mention the `NodeInterrupt`?

Reply:
yes,i think it's used likely by some users?